allennlp/predictors/dialog_qa.py

```diff
@@ -0,0 +1,56 @@
+import json
+from overrides import overrides
+
+from allennlp.common.util import JsonDict
+from allennlp.data import DatasetReader, Instance
+from allennlp.data.tokenizers.word_splitter import SpacyWordSplitter
+from allennlp.predictors.predictor import Predictor
+from allennlp.models import Model
+
+
+@Predictor.register('dialog_qa')
+class DialogQAPredictor(Predictor):
+    def __init__(self, model: Model, dataset_reader: DatasetReader) -> None:
+        super().__init__(model, dataset_reader)
+        self._tokenizer = SpacyWordSplitter(language='en_core_web_sm')
+
+    def predict(self, jsonline: str) -> JsonDict:
+        """
+        Make a machine comprehension prediction on the supplied input.
+       -------
+        A dictionary that represents the prediction made by the system.  The answer string will be under the
+        "best_span_str" key.
+        """
+        return self.predict_json(json.loads(jsonline))
+
+    @overrides
+    def _json_to_instance(self, json_dict: JsonDict) -> Instance:
+        """
+        Expects json that looks like the original quac file
+        """
+        paragraph_json = json_dict["paragraphs"][0]
+        paragraph = paragraph_json['context']
+        tokenized_paragraph = self._tokenizer.split_words(paragraph)
+        qas = paragraph_json['qas']
+        metadata = {}
+        metadata["instance_id"] = [qa['id'] for qa in qas]
+        question_text_list = [qa["question"].strip().replace("\n", "") for qa in qas]
+        answer_texts_list = [[answer['text'] for answer in qa['answers']] for qa in qas]
```

Comment:
You also probably shouldn't assume that you always have the answer given.

Reply:
The answers to the previous questions in the dialog is actually *input* to the later questions. So it's kind of necessary to have it here. The more realistic demo scenario would allow interaction back and forth between user / model will be much more challenging to implement (i think?), and the purpose of this predictor is not necessarily for demo (because of that reason), but to allow outputting to the json file after the predictions.

Reply:
Hmm, right, ok.  But, your last statement makes me think there's something wrong with our API, because you shouldn't have to implement this just to get JSON predictions out.  I'm going to open an issue for that.

Reply:
Fixed. Thanks!